### PR TITLE
feat(c11): Add -wextra flag to compiler options

### PR DESCRIPTION
**Pull request by miguelafsilva5**
The -wextra flag adds several warning/error sources that are not encompassed in -wall.
This PR adds this flag and adapts the code to comply with it.

This PR depends on PR #122 

The code on this PR was validated in the the following platforms:
- qemu_aarch64_virt
- qemu_riscv64_virt

### CODE REVIEW

File: src/arch/armv8/armv8-r/mpu.c [hunk 1]
```diff
@@ -157,7 +157,7 @@ static inline mem_attrs_t mpu_entry_attrs(struct mp_region* mpr)
 static mpid_t mpu_entry_allocate()
 {
     mpid_t reg_num = INVALID_MPID;
-    for (mpid_t i = 0; i < mpu_num_entries(); i++) {
+    for (mpid_t i = 0; i < (ssize_t)mpu_num_entries(); i++) {
```
Comment (josecm) [inline src/arch/armv8/armv8-r/mpu.c:160]: Maybe we could cast this to `mpid_t`?

File: src/arch/armv8/armv8-r/mpu.c [hunk 2]
```diff
@@ -546,7 +546,7 @@ void mpu_init()
     bitmap_clear_consecutive(cpu()->arch.profile.mpu.bitmap, 0, mpu_num_entries());
     list_init(&cpu()->arch.profile.mpu.order.list);
 
-    for (mpid_t mpid = 0; mpid < mpu_num_entries(); mpid++) {
+    for (mpid_t mpid = 0; mpid < (ssize_t)mpu_num_entries(); mpid++) {
```
Comment (josecm) [inline src/arch/armv8/armv8-r/mpu.c:549]: The same here.

File: src/arch/armv8/vgic.c [hunk 3]
```diff
@@ -779,35 +819,22 @@ struct vgic_reg_handler_info ipriorityr_info = {
     vgic_int_set_prio_hw,
 };
 
-struct vgic_reg_handler_info vgicd_misc_info = {
-    vgicd_emul_misc_access,
-    0x4,
-};
+struct vgic_reg_handler_info vgicd_misc_info = { vgicd_emul_misc_access, 0x4, 0, 0, 0, NULL, NULL,
+    NULL };
```
Comment (josecm) [inline src/arch/armv8/vgic.c:823]: We can add a comma after the last `NULL` to force the formatted to put one field in each line.

Comment (josecm) [reply src/arch/armv8/vgic.c:823]: The same for the rest vgic info structs..

File: src/arch/armv8/vgicv3.c [hunk 4]
```diff
@@ -176,18 +191,12 @@ struct vgic_reg_handler_info irouter_info = {
...
-};
-struct vgic_reg_handler_info vgicr_typer_info = {
-    vgicr_emul_typer_access,
-    0xC,
-};
-struct vgic_reg_handler_info vgicr_pidr_info = {
-    vgicr_emul_pidr_access,
-    0x4,
-};
+struct vgic_reg_handler_info vgicr_ctrl_info = { vgicr_emul_ctrl_access, 0x4, 0, 0, 0, NULL, NULL,
```
Comment (josecm) [inline src/arch/armv8/vgicv3.c:194]: Same here.

**Review by josecm — CHANGES_REQUESTED** [review]

**Review by DavidMCerdeira — APPROVED** [review]

File: Makefile [hunk 5]
```diff
@@ -189,8 +189,10 @@ ifeq ($(DEBUG), y)
 	OPTIMIZATIONS:=g
 endif
 
-override CFLAGS+=-O$(OPTIMIZATIONS) -Wall -Werror -ffreestanding -std=c11 -pedantic -pedantic-errors \
-	-fno-pic $(arch-cflags) $(platform-cflags) $(CPPFLAGS) $(debug_flags)
+
```
Comment (DavidMCerdeira) [inline Makefile:192]: Is it better with extra space?

File: src/arch/armv8/armv8-r/inc/arch/mem.h [hunk 6]
```diff
@@ -52,7 +52,7 @@ typedef union {
```
Comment (DavidMCerdeira) [inline src/arch/armv8/armv8-r/inc/arch/mem.h:52]: Commit 4643b407738578911f02967ded07e2367e8c33ee description:
(-Werror=old-style-declaration) Additionalloy,
the 'const' keyword is not need in static inline functions.

Not sure how the info in parenthesis relates to the rest.
'Additionalloy' is not correctly spelled

File: src/arch/armv8/armv8-r/mpu.c (hunk 1 above)
Comment (miguelafsilva5) [reply src/arch/armv8/armv8-r/mpu.c]: Fixed in aef4397dde5adfee49fdfa9159b05c1b1ccfd493

File: src/arch/armv8/armv8-r/mpu.c (hunk 2 above)
Comment (miguelafsilva5) [reply src/arch/armv8/armv8-r/mpu.c]: Fixed in aef4397dde5adfee49fdfa9159b05c1b1ccfd493

File: Makefile (hunk 5 above)
Comment (miguelafsilva5) [reply Makefile:192]: The next PR in the chain modifies those lines. It is properly formatted there.

File: src/arch/armv8/vgic.c (hunk 3 above)
Comment (miguelafsilva5) [reply src/arch/armv8/vgic.c]: Fixed in 2d0aea7612bcb21d087afdf77c66b2e835a0061b

File: src/arch/armv8/armv8-r/inc/arch/mem.h (hunk 6 above)
Comment (miguelafsilva5) [reply src/arch/armv8/armv8-r/inc/arch/mem.h:52]: Fixed in commit d17aa5ff754f3798b49971a04d47421931ad43c7

File: src/arch/armv8/vgicv3.c (hunk 4 above)
Comment (miguelafsilva5) [reply src/arch/armv8/vgicv3.c]: There were some missing in vgicv2.c too. Both files should be correct in commit 13014b264a97b1a89b9e9a4e75cd2f9865d289d9

**Review by DavidMCerdeira — APPROVED** [review]

**Review by josecm — APPROVED** [review]

**Pull request merged**